### ISSUE TITLE
IP targets broken for FARGATE tasks in v2.2.2 (eks k8s 1.19, 1.20)

### ISSUE THREAD

**Issue by luzhkovvv** [kubernetes-sigs/aws-load-balancer-controller#2156]
**Describe the bug**
When creating ingress with "target-type: ip" , FARGATE-based tasks fails to register in target groups with "Reconciler error"

**Steps to reproduce**
Within a namespace, associated with FARGATE profile, create sample deployment/service/ingress:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx
        resources:
          limits:
            memory: "128Mi"
            cpu: "500m"
        ports:
        - containerPort: 80
```
```
apiVersion: v1
kind: Service
metadata:
  name: nginx
spec:
  type: ClusterIP
  clusterIP: None
  selector:
    app: nginx
  ports:
  - port: 80
```
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: nginx
  labels:
    name: nginx
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
spec:
  rules:
  - host: nginx.local
    http:
      paths:
      - pathType: Prefix
        path: "/"
        backend:
          service:
            name: nginx
            port: 
              number: 80
```

**Expected outcome**
no errors, pod IP is registered as a target in a target group, controller log contains:
```
{
    "level": "info",
    "ts": 1628020056.8204412,
    "msg": "registering targets",
    "arn": "arn:aws:elasticloadbalancing:eu-west-1:XXX:targetgroup/k8s-default-nginx-c6d99ceb1e/b97d23e42f4e7d0a",
    "targets": [
        {
            "AvailabilityZone": null,
            "Id": "10.104.90.11",
            "Port": 80
        }
    ]
}
{
    "level": "info",
    "ts": 1628020057.1608725,
    "msg": "registered targets",
    "arn": "arn:aws:elasticloadbalancing:eu-west-1:XXX:targetgroup/k8s-default-nginx-c6d99ceb1e/b97d23e42f4e7d0a"
}
```

**Actual outcome**
target group's target list remains empty, AWS load balancer controller log contains:
```
{
    "level": "error",
    "ts": 1628020370.6876273,
    "logger": "controller",
    "msg": "Reconciler error",
    "reconcilerGroup": "elbv2.k8s.aws",
    "reconcilerKind": "TargetGroupBinding",
    "controller": "targetGroupBinding",
    "name": "k8s-fg-nginx-529005d44e",
    "namespace": "fg",
    "error": "InvalidInstanceID.Malformed: Invalid id: \"fargate-ip-10-104-92-107.eu-west-1.compute.internal\"\n\tstatus code: 400, request id: 107bef5a-9683-4f64-94b6-69b8468c4285"
}
```
meanwhile, the target group is correctly created with IP target type

**Environment**

* AWS Load Balancer controller version - 2.2.2 deployed by helm chart 1.2.5
* Kubernetes version - 1.19, 1.20
* Using EKS (yes/no), if so version? - yes, nodes v1.20.4-eks-6b7464, v1.19.6-eks-e91815, eks 1.20 eks.2, 1.19 eks.dontremember

**Additional Context**:
The same definitions work fine on the exact same cluster with the same (2.2.2) controller version within a nodegroup-backed namespace (same node version, same VPC, etc) - expected outcome logs are from that test.
The same definitions work fine on exact same cluster and the exact same fargate-backed namespace with 2.2.1 or 2.1.3 controller versions, deployed by official helm charts 1.2.3 or 1.1.6.
Current k8s version which I used for testing is 1.20, but same behaviour was observed earlier with k8s 1.19.
For me, it seems that the 2.2.2 controller when used with IP target type and fargate pod tries to register in the target group fargate instance instead of registering an IP address.

**Comment by M00nF1sh** [comment]
@luzhkovvv 
Sorry for the trouble caused. I have updated the release page of v2.2.2 to call out this issue.
Fix is WIP we'll have a new patch version out soon: https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/2158

**Comment by M00nF1sh** [comment]
@luzhkovvv v2.2.3 is released, which now works for Fargate as well.

**Comment by kishorj** [comment]
Fix is available in the v2.2.3 release. Closing the issue.
